core/src/components/toggle/test/bottom-content/toggle.e2e.ts

```diff
@@ -0,0 +1,196 @@
+      const helperText = page.locator('ion-toggle .helper-text');
+      const helperTextId = await helperText.getAttribute('id');
+      const ariaDescribedBy = await input.getAttribute('aria-describedby');
+
+      expect(ariaDescribedBy).toBe(helperTextId);
+    });
+    test('error text should be visible when toggle is invalid', async ({ page }) => {
+      await page.setContent(
+        `<ion-toggle class="ion-invalid ion-touched" helper-text="Helper text" error-text="Error text">Label</ion-toggle>`,
+        config
+      );
+
+      const helperText = page.locator('ion-toggle .helper-text');
+      const errorText = page.locator('ion-toggle .error-text');
+      await expect(helperText).toBeHidden();
+      await expect(errorText).toBeVisible();
+      await expect(errorText).toHaveText('Error text');
+    });
+
+    test('input should have an aria-describedby attribute when error text is present', async ({ page }) => {
+      await page.setContent(
+        `<ion-toggle class="ion-invalid ion-touched" helper-text="Helper text" error-text="Error text">Label</ion-toggle>`,
+        config
+      );
+
+      const input = page.locator('ion-toggle input[type=checkbox]');
+      const errorText = page.locator('ion-toggle .error-text');
+      const errorTextId = await errorText.getAttribute('id');
+      const ariaDescribedBy = await input.getAttribute('aria-describedby');
+
+      expect(ariaDescribedBy).toBe(errorTextId);
+    });
+    test('input should have aria-invalid attribute when toggle is invalid', async ({ page }) => {
+      await page.setContent(
+        `<ion-toggle class="ion-invalid ion-touched" helper-text="Helper text" error-text="Error text">Label</ion-toggle>`,
+        config
+      );
+
+      const input = page.locator('ion-toggle input[type=checkbox]');
+
+      await expect(input).toHaveAttribute('aria-invalid');
+    });
+    test('input should not have aria-invalid attribute when toggle is valid', async ({ page }) => {
+      await page.setContent(`<ion-toggle helper-text="Helper text" error-text="Error text">Label</ion-toggle>`, config);
+
+      const input = page.locator('ion-toggle input[type=checkbox]');
+
+      await expect(input).not.toHaveAttribute('aria-invalid');
+    });
+    test('input should not have aria-describedby attribute when no hint or error text is present', async ({ page }) => {
+      await page.setContent(`<ion-toggle>Label</ion-toggle>`, config);
+
+      const input = page.locator('ion-toggle input[type=checkbox]');
+
+      await expect(input).not.toHaveAttribute('aria-describedby');
+    });
+  });
+});
+
+/**
+ * Rendering is different across modes
+ */
+configs({ modes: ['ios', 'md'], directions: ['ltr'] }).forEach(({ title, screenshot, config }) => {
+  test.describe(title('toggle: helper text rendering'), () => {
+    // Check the default label placement, end, and stacked
+    [undefined, 'end', 'stacked'].forEach((labelPlacement) => {
```

Comment:
What was the reasoning for not checking `fixed`?

Reply:
[Answered](https://github.com/ionic-team/ionic-framework/pull/30140#discussion_r1962445250)